services/spar/src/Spar/Scim.hs

```diff
@@ -264,60 +284,82 @@ updateValidSCIMUser tokinfo uidText newScimUser = do
 
 
-toSCIMStoredUser
+toScimStoredUser
   :: forall m. (SAML.HasNow m, MonadReader Env m)
-  => UserId -> SCIM.User.User -> m SCIM.Class.User.StoredUser
-toSCIMStoredUser uid usr = do
+  => UserId -> Scim.User.User -> m Scim.Class.User.StoredUser
+toScimStoredUser uid usr = do
   now <- SAML.getNow
   baseuri <- asks $ derivedOptsScimBaseURI . derivedOpts . sparCtxOpts
-  pure $ toSCIMStoredUser' now baseuri uid usr
+  pure $ toScimStoredUser' now baseuri uid usr
 
-toSCIMStoredUser'
+toScimStoredUser'
   :: HasCallStack
-  => SAML.Time -> URIBS.URI -> UserId -> SCIM.User.User -> SCIM.Class.User.StoredUser
-toSCIMStoredUser' (SAML.Time now) baseuri (idToText -> uid) usr = SCIM.WithMeta meta thing
+  => SAML.Time -> URIBS.URI -> UserId -> Scim.User.User -> Scim.Class.User.StoredUser
+toScimStoredUser' (SAML.Time now) baseuri (idToText -> uid) usr =
+    Scim.WithMeta meta (Scim.WithId uid usr)
   where
     mkLocation :: String -> URI
     mkLocation pathSuffix = convURI $ baseuri SAML.=/ cs pathSuffix
       where
         convURI uri = fromMaybe err . parseURI . cs . URIBS.serializeURIRef' $ uri
           where err = error $ "internal error: " <> show uri
 
-    thing = SCIM.WithId uid usr
-    thingHash = hashlazy (Aeson.encode thing) :: Digest SHA256
-
-    meta = SCIM.Meta
-      { SCIM.resourceType = SCIM.UserResource
-      , SCIM.created = now
-      , SCIM.lastModified = now
-      , SCIM.version = SCIM.Strong (Text.pack (show thingHash))
-        -- TODO: replace SCIM.Strong with SCIM.Weak
-        -- TODO: it looks like (a) we need to add this to the HTTP header.
+    meta = Scim.Meta
+      { Scim.resourceType = Scim.UserResource
+      , Scim.created = now
+      , Scim.lastModified = now
+      , Scim.version = calculateVersion uid usr
+        -- TODO: it looks like we need to add this to the HTTP header.
         -- https://tools.ietf.org/html/rfc7644#section-3.14
-        -- TODO: scim versions are called versions, but they aren't really monotonously
-        -- increasing.  add a note to hscim?
-      , SCIM.location = SCIM.URI . mkLocation $ "/Users/" <> cs uid
+      , Scim.location = Scim.URI . mkLocation $ "/Users/" <> cs uid
       }
 
-updSCIMStoredUser
+updScimStoredUser
   :: forall m. (SAML.HasNow m)
-  => SCIM.User.User -> SCIM.Class.User.StoredUser -> m SCIM.Class.User.StoredUser
-updSCIMStoredUser usr storedusr = do
+  => Scim.User.User -> Scim.Class.User.StoredUser -> m Scim.Class.User.StoredUser
+updScimStoredUser usr storedusr = do
   now <- SAML.getNow
-  pure $ updSCIMStoredUser' now usr storedusr
-
--- TODO: recalculate the 'version', too
-updSCIMStoredUser'
-  :: SAML.Time -> SCIM.User.User -> SCIM.Class.User.StoredUser -> SCIM.Class.User.StoredUser
-updSCIMStoredUser' (SAML.Time moddate) usr (SCIM.WithMeta meta (SCIM.WithId scimuid _)) =
-  SCIM.WithMeta (meta { SCIM.lastModified = moddate }) (SCIM.WithId scimuid usr)
+  pure $ updScimStoredUser' now usr storedusr
+
+updScimStoredUser'
```

Comment:
`updSCIMStoredUser'` wasn't updating the version, which was clearly a bug; now it does.